Firestore/core/src/model/target_index_matcher.h

```diff
@@ -88,10 +89,21 @@ class TargetIndexMatcher {
   bool MatchesOrderBy(const core::OrderBy& order_by,
                       const model::Segment& segment);
 
+  // Custom comparator for FieldFilter based on its Field property.
```

Comment:
a custom comparator to add FieldFilters into a set based on their FieldPath only.

Reply:
This set is too specific to this scenario, differentiating FieldFilter based on its field path only. This should not be a generic case for FieldFilter comparison.

Reply:
Agreed